src/generators/Generator.js

```diff
@@ -260,14 +260,26 @@ export default class Generator {
 					// export is last property, we can just return it
 					this.code.overwrite( defaultExport.start, defaultExport.declaration.start, `return ` );
 				} else {
-					// TODO ensure `template` isn't already declared
-					this.code.overwrite( defaultExport.start, defaultExport.declaration.start, `var template = ` );
+					// we need to avoid a conflict with anything on the top-level scope of the component
+					// however, determining which things these are is tricky, so we instead just avoid conflicts with any identifiers anywhere
```

Comment:
I'm calling `annotateWithScope` on the entire AST, and it looks like it runs into problems [here](https://github.com/sveltejs/svelte/blob/master/src/generators/annotateWithScopes.js#L63) when it reaches the `ExportDefaultDeclaration` node - `node.id` is not defined. Is it safe just to add an extra test for whether `node.id` is present before noting that `node.id.name` is a declaration?